cmd/env.go

```diff
@@ -361,3 +337,38 @@ func commonEnvFlags(flags *pflag.FlagSet) (server, namespace, context string, er
 
 	return server, namespace, context, nil
 }
+
+func resolveEnvFlags(flags *pflag.FlagSet) (server, namespace string, err error) {
+	server, ns, ctx, err := commonEnvFlags(flags)
+	if err != nil {
+		return "", "", err
+	}
+
+	if len(server) == 0 {
+		// If server is not provided, use the provided context.
+		// If context is also not provided, use the current context.
+		var context *string
+		if len(ctx) != 0 {
+			context = &ctx
```

Comment:
This block of code seems odd! We're setting `context = &ctx` so that `resolveContext` can then do a similar dance and set `ctxName = *context`. I feel like I'm on the edge of being able to understand this code at all. Do you think it's possible to simplify this so that it's less ... intricate?

Reply:
I've cleaned this up by separating `resolveContext` into 

1. `resolveContext()` which resolves based on the default context and
2. `resolveContextFor(context string)` which resolves based on the context in the argument.

That should hopefully clear things up.